dev-tools/mage/integtest.go

```diff
@@ -185,37 +185,50 @@ func NewIntegrationRunners(path string, passInEnv map[string]string) (Integratio
 		if err != nil {
 			return nil, errors.Wrapf(err, "%s tester failed on Use", t.Name())
 		}
-		if use {
-			// Create the steps for the specific runner.
-			var runnerSteps IntegrationTestSteps
-			requirements := t.StepRequirements()
-			if requirements != nil {
-				runnerSteps = append(runnerSteps, requirements...)
-			}
-			runnerSteps = append(runnerSteps, steps...)
+		if !use {
+			continue
+		}
+		// Create the steps for the specific runner.
+		var runnerSteps IntegrationTestSteps
+		requirements := t.StepRequirements()
+		if requirements != nil {
+			runnerSteps = append(runnerSteps, requirements...)
+		}
+		runnerSteps = append(runnerSteps, steps...)
 
-			// Create the custom env for the runner.
-			env := map[string]string{}
-			for k, v := range passInEnv {
-				env[k] = v
-			}
-			env[insideIntegrationTestEnvVar] = "true"
-			passThroughEnvs(env, defaultPassthroughEnvVars...)
-			if mg.Verbose() {
-				env["MAGEFILE_VERBOSE"] = "1"
-			}
-			if UseVendor {
-				env["GOFLAGS"] = "-mod=vendor"
-			}
+		// Create the custom env for the runner.
+		env := map[string]string{}
+		for k, v := range passInEnv {
+			env[k] = v
+		}
+		env[insideIntegrationTestEnvVar] = "true"
+		passThroughEnvs(env, defaultPassthroughEnvVars...)
+		if mg.Verbose() {
+			env["MAGEFILE_VERBOSE"] = "1"
+		}
+		if UseVendor {
+			env["GOFLAGS"] = "-mod=vendor"
+		}
 
-			runner := &IntegrationRunner{
-				steps:  runnerSteps,
-				tester: t,
-				dir:    dir,
-				env:    env,
-			}
-			runners = append(runners, runner)
+		runner := &IntegrationRunner{
+			steps:  runnerSteps,
+			tester: t,
+			dir:    dir,
+			env:    env,
+		}
+		runners = append(runners, runner)
+	}
+	// Keep support for modules that don't have a local environment defined at the module
+	// level (system, stack and cloud modules by now)
+	if len(runners) == 0 {
+		if mg.Verbose() {
+			fmt.Printf(">> No runner found in %s, using docker\n", path)
+		}
+		runner, err := NewDockerIntegrationRunner("MODULE")
```

Comment:
Your going to need to pass through more than just MODULE, and MODULE is not going to be set in the current process environment, as its now passed through to `NewIntegrationRunners`.

Probably would be good to change `NewDockerIntegrationRunner` to also take a `passInEnv map[string]string` so you can:

`NewDockerIntegrationRunner(passInEnv)`

Reply:
Good point, I have refactored runner initialization so same code is used in the three places, and it can receive variables as an env map or as a list of pass through variable names.